controllers/storageclaim_controller.go

```diff
@@ -613,3 +643,103 @@ func (r *StorageClassClaimReconciler) createOrReplaceVolumeSnapshotClass(volumeS
 	}
 	return nil
 }
+
+func (r *StorageClaimReconciler) migrateStorageClassClaims() error {
+	storageClassClaimsList := &v1alpha1.StorageClassClaimList{}
+	if err := r.list(storageClassClaimsList); err != nil {
+		return fmt.Errorf("failed to list storageclassclaims")
+	}
+
+	for idx := range storageClassClaimsList.Items {
+		storageClassClaim := &storageClassClaimsList.Items[idx]
+		storageClaim := &v1alpha1.StorageClaim{}
+		storageClaim.Name = storageClassClaim.Name
+
+		// migrate storageclassclaim to storageclaim
+		_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageClaim, func() error {
+			storageClaim.Spec.Type = storageClassClaim.Spec.Type
+			storageClaim.Spec.EncryptionMethod = storageClassClaim.Spec.EncryptionMethod
+			storageClaim.Spec.StorageProfile = storageClassClaim.Spec.StorageProfile
+			storageClaim.Spec.StorageClient = storageClassClaim.Spec.StorageClient.DeepCopy()
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("failed to copy spec from storageclassclaims to storageclaims: %v", err)
+		}
+
+		for idx := range storageClassClaim.Status.SecretNames {
+			secretName := storageClassClaim.Status.SecretNames[idx]
+			secret := &corev1.Secret{}
+			secret.Name = secretName
+			secret.Namespace = storageClassClaim.Spec.StorageClient.Namespace
+			if err = r.get(secret); err != nil {
+				return fmt.Errorf("failed to get secret %s: %v", client.ObjectKeyFromObject(secret), err)
+			}
+
+			if err = controllerutil.SetOwnerReference(storageClaim, secret, r.Scheme); err != nil {
+				return fmt.Errorf("failed to set owner reference on secret %s: %v", client.ObjectKeyFromObject(secret), err)
+			}
+		}
```

Comment:
Why are we migrating existing secrets? The new secrets will get reconciled as of the even of the creation of the new StorageClaim

Reply:
I've a choice b/n migrating existing secrets and set owner ref so that storageclaim cleans them up or explicitly delete existing secret and let new secret be populated by storageclaim.

I chose the former, unless there's an objection (or any case that I missed) I want to stick w/ current sol.

Reply:
The secrets now have new names so they are different anyways. Lets just delete the old one and let the reconciler create the new ones

Reply:
Will do.

Reply:
refactored. done.